libp2p/multicodec.nim

```diff
@@ -251,37 +257,32 @@ const
 
 proc multiCodec*(name: string): MultiCodec {.compileTime.} =
   ## Generate MultiCodec from string ``name`` at compile time.
-  var code = NameCodecs.getOrDefault(name, -1)
-  if code == -1:
-    raise newException(MultiCodecError,
-                       "MultiCodec `" & name & "` not supported!")
-  result = MultiCodec(code)
+  let code = NameCodecs.getOrDefault(name, -1)
+  doAssert(code != -1)
+  MultiCodec(code)
 
 proc multiCodec*(code: int): MultiCodec {.compileTime.} =
   ## Generate MultiCodec from integer ``code`` at compile time.
-  var name = CodeCodecs.getOrDefault(code, "")
-  if name == "":
-    raise newException(MultiCodecError,
-                       "MultiCodec with code " & $code & " not supported!")
-  result = MultiCodec(code)
+  let name = CodeCodecs.getOrDefault(code, "")
+  doAssert(name != "")
+  MultiCodec(code)
 
 proc `$`*(mc: MultiCodec): string =
   ## Returns string representation of MultiCodec ``mc``.
-  result = CodeCodecs.getOrDefault(int(mc), "")
-  if result == "":
-    raise newException(MultiCodecError,
-                       "MultiCodec with code " & $int(mc) & " not supported!")
+  let name = CodeCodecs.getOrDefault(int(mc), "")
+  doAssert(name != "")
```

Comment:
Maybe remove this and throw like before?

Reply:
just have it return "Unsupported codec X"

Reply:
You can safely return an empty string here.